feature-libs/quote/components/actions/by-role/quote-actions-by-role.component.spec.ts

```diff
@@ -0,0 +1,714 @@
+      allowedActions: [
+        { type: QuoteActionType.SUBMIT, isPrimary: true },
+        { type: QuoteActionType.CANCEL, isPrimary: false },
+        { type: QuoteActionType.EDIT, isPrimary: false },
+      ],
+      state: QuoteState.BUYER_OFFER,
+    };
+    const confirmationContextForEditAction: ConfirmationContext = {
+      quote: quoteInBuyerOfferState,
+      title: 'quote.actions.confirmDialog.buyer_offer.edit.title',
+      confirmNote: 'quote.actions.confirmDialog.buyer_offer.edit.confirmNote',
+      warningNote: 'quote.actions.confirmDialog.buyer_offer.edit.warningNote',
+      validity: 'quote.actions.confirmDialog.validity',
+    };
+    mockQuoteDetails$.next(quoteInBuyerOfferState);
+    fixture.detectChanges();
+    component.onClick(
+      QuoteActionType.EDIT,
+      quoteInBuyerOfferState,
+      currentCart
+    );
+    expect(launchDialogService.openDialog).toHaveBeenCalledWith(
+      LAUNCH_CALLER.ACTION_CONFIRMATION,
+      component.element,
+      component['viewContainerRef'],
+      { confirmationContext: confirmationContextForEditAction }
+    );
+  });
+
+  it('should not open confirmation dialog when action is CANCEL and state is BUYER_DRAFT', () => {
+    spyOn(launchDialogService, 'openDialog');
+    const quoteInBuyerDraftState: Quote = {
+      ...mockQuote,
+      allowedActions: [
+        { type: QuoteActionType.SUBMIT, isPrimary: true },
+        { type: QuoteActionType.CANCEL, isPrimary: false },
+        { type: QuoteActionType.EDIT, isPrimary: false },
+      ],
+      state: QuoteState.BUYER_DRAFT,
+    };
+    mockQuoteDetails$.next(quoteInBuyerDraftState);
+    fixture.detectChanges();
+    component.onClick(
+      QuoteActionType.CANCEL,
+      quoteInBuyerDraftState,
+      currentCart
+    );
+    expect(launchDialogService.openDialog).toHaveBeenCalledTimes(0);
+  });
+
+  it('should open confirmation dialog when action is REQUOTE and state is EXPIRED', () => {
+    spyOn(launchDialogService, 'openDialog');
+    const expiredQuote: Quote = {
+      ...mockQuote,
+      allowedActions: [{ type: QuoteActionType.REQUOTE, isPrimary: true }],
+      state: QuoteState.EXPIRED,
+    };
+    const confirmationContextForRequoteAction: ConfirmationContext = {
+      quote: expiredQuote,
+      title: 'quote.actions.confirmDialog.expired.requote.title',
+      confirmNote: 'quote.actions.confirmDialog.expired.requote.confirmNote',
+      warningNote: 'quote.actions.confirmDialog.expired.requote.warningNote',
+    };
+    mockQuoteDetails$.next(expiredQuote);
+    fixture.detectChanges();
+
+    component.onClick(QuoteActionType.REQUOTE, expiredQuote, currentCart);
+    expect(launchDialogService.openDialog).toHaveBeenCalledWith(
+      LAUNCH_CALLER.ACTION_CONFIRMATION,
+      component.element,
+      component['viewContainerRef'],
+      { confirmationContext: confirmationContextForRequoteAction }
+    );
+  });
+
+  describe('Threshold check', () => {
+    const attributeName = 'disabled';
```

Comment:
@Larisa-Staroverova: 'attributeName' is not a good name.
personally i would instead simply inline this constant, because the string literal 'disabled' is already very descriptive.

Reply:
Done!